demo/site/src/documents/pages/Page.tsx

```diff
@@ -76,12 +77,15 @@ async function fetchData({ pageTreeNodeId, scope }: Props) {
 
 export async function generateMetadata({ pageTreeNodeId, scope }: Props, parent: ResolvingMetadata): Promise<Metadata> {
     const data = await fetchData({ pageTreeNodeId, scope });
+    const siteConfig = getSiteConfigForDomain(scope.domain);
+
     const document = data?.pageContent?.document;
     if (!document) {
         return {};
     }
-    const siteUrl = "http://localhost:3000"; //TODO get from site config
-    const canonicalUrl = document.seo.canonicalUrl || `${siteUrl}${data.pageContent.path}`;
+
+    const siteUrl = siteConfig.url;
+    const canonicalUrl = document.seo.canonicalUrl || `${siteUrl}${data.pageContent.path}${scope.language}`;
```

Comment:
```suggestion
    const canonicalUrl = document.seo.canonicalUrl || `${siteUrl}${data.pageContent.path}/${scope.language}`;
```

Do we need a slash here?

Reply:
language comes first:
```suggestion
    const canonicalUrl = document.seo.canonicalUrl || `${siteUrl}/${scope.language}${data.pageContent.path}`;
```

Reply:
@johnnyomair nice catch. You found a bug in one of our projects :)

Reply:
Now it works for the homepage and for child pages
<img width="631" alt="Screenshot 2025-05-20 at 11 12 21" src="https://github.com/user-attachments/assets/afa97703-d6be-4e80-bd4e-f7ecc64a137a" />
<img width="630" alt="Screenshot 2025-05-20 at 11 09 49" src="https://github.com/user-attachments/assets/135f4fb7-4666-48e8-94ef-6887dafa4e7d" />